File: services/frontend/src/lib/api/eakten/types/eakten-application-mapping.ts
interface EaktenApplicationMappingModel {
  id: string;
  application_id: string;
  eakte_akte_id: string;
  application: EaktenApplicationMappingApplicationShortModel;
  applicant: EaktenApplicationMappingApplicantShortModel;
  school_degree: EaktenApplicationMappingDegreeShortModel;
  school: EaktenApplicationMappingSchoolShortModel;
  eakte_akte: EaktenApplicationMappingAkteShortModel;
}

interface EaktenApplicationMappingApplicationShortModel {
  id: string;
  status: EaktenApplicationMappingApplicationStatusShortModel;
}

interface EaktenApplicationMappingApplicationStatusShortModel {
  id: string;
  identifier: string;
  name: string;
}

interface EaktenApplicationMappingApplicantShortModel {
  id: string;
  firstname: string;
  lastname: string;
}

interface EaktenApplicationMappingDegreeShortModel {
  id: string;
  name: string;
}

interface EaktenApplicationMappingSchoolShortModel {
  id: string;
  name: string;
}

interface EaktenApplicationMappingAkteShortModel {
  id: string;
  aktenzeichen: string;
}

File: services/backend/src/resources/eakte/service/models/eakten_application_mapping.go
package models

import "github.com/google/uuid"

type EaktenApplicationMappingModel struct {
	ID               uuid.UUID
	ApplicationID    uuid.UUID
	EakteAkteID      uuid.UUID
	Application      *EaktenApplicationMappingApplicationShortModel
	Applicant        *EaktenApplicationMappingApplicantShortModel
	SchoolWithDegree *EaktenApplicationMappingDegreeShortModel
	School           *EaktenApplicationMappingSchoolShortModel
	Eakte_Akte       *EaktenApplicationMappingAkteShortModel
}

type EaktenApplicationMappingApplicationShortModel struct {
	ID     uuid.UUID
	Status *EaktenApplicationMappingApplicationStatusShortModel
}

type EaktenApplicationMappingApplicationStatusShortModel struct {
	ID         uuid.UUID
	Identifier string
	Name       string
}

type EaktenApplicationMappingApplicantShortModel struct {
	ID        uuid.UUID
	FirstName string
	LastName  string
}

type EaktenApplicationMappingDegreeShortModel struct {
	ID   uuid.UUID
	Name string
}

type EaktenApplicationMappingSchoolShortModel struct {
	ID   uuid.UUID
	Name string
}

type EaktenApplicationMappingAkteShortModel struct {
	ID           uuid.UUID
	Aktenzeichen string
}
